src/plugins/dashboard/public/dashboard_container/external_api/dashboard_renderer.tsx

```diff
@@ -133,7 +144,9 @@ export const DashboardRenderer = forwardRef<AwaitingDashboardAPI, DashboardRende
 
     return (
       <div className={viewportClasses}>
-        {loading ? loadingSpinner : <div ref={dashboardRoot} />}
+        {loading && loadingSpinner}
```

Comment:
nit, there should never be a case where loadingSpinner and fatalError.render() are displayed but there could be if both loading and fatalError are set. Would it be clearer to move this logic to a function and use early returns so only a single item is rendered. For example
```
function renderContent() {
  if (loading) {
    return loadingSpinner;
  }

  if (fatalError) {
    return fatalError.render();
  }

  return (<div ref={dashboardRoot} />);
}
```

Reply:
That's true. Even if it wouldn't necessarily happen, it's still much cleaner and more understandable to do it this way.